test/unit/Main.hs

```diff
@@ -627,8 +639,84 @@ inventory =
             (hash (E.insertCount 5 x E.empty))
             (hash (E.union (E.insertCount 2 x E.empty) (E.insertCount 3 x E.empty)))
         )
+    , testCase
+        "difference"
+        ( assertEqual
+            "{(2,x),(3,y)} difference {(3,x),(1,y)} = {(0,x), (2,y)}"
+            ( hash
+                ( E.insertCount 2 x (E.insertCount 3 y E.empty)
+                    `E.difference` E.insertCount 3 x (E.insertCount 1 y E.empty)
+                )
+            )
+            (hash (E.insertCount 0 x (E.insertCount 2 y E.empty)))
+        )
+    , testCase
+        "subset / yes"
+        ( assertBool
+            "{(0,x),(3,y),(2,z)} isSubsetOf {(3,y),(4,z)}"
+            ( E.insertCount 0 x (E.insertCount 3 y (E.insertCount 2 z E.empty))
+                `E.isSubsetOf` E.insertCount 3 y (E.insertCount 4 z E.empty)
+            )
+        )
+    , testCase
+        "subset / no"
+        ( assertBool
+            "{(2,x),(3,y)} isSubsetOf {(1,x),(4,y)}"
+            ( not
+                ( E.insertCount 2 x (E.insertCount 3 y E.empty)
+                    `E.isSubsetOf` E.insertCount 1 x (E.insertCount 4 y E.empty)
+                )
+            )
+        )
+    , testCase
+        "isEmpty / yes"
+        ( assertBool
+            "isEmpty {(0,x),(0,y)}"
+            (E.isEmpty (E.insertCount 0 x (E.insertCount 0 y E.empty)))
+        )
+    , testCase
+        "isEmpty / no"
+        ( assertBool
+            "isEmpty {(0,x),(1,y)}"
+            (not (E.isEmpty (E.insertCount 0 x (E.insertCount 1 y E.empty))))
+        )
     ]
  where
   x = E.mkEntity (defaultEntityDisplay 'X') "fooX" [] [] []
   y = E.mkEntity (defaultEntityDisplay 'Y') "fooY" [] [] []
-  _z = E.mkEntity (defaultEntityDisplay 'Z') "fooZ" [] [] []
+  z = E.mkEntity (defaultEntityDisplay 'Z') "fooZ" [] [] []
+
+misc :: TestTree
+misc =
+  testGroup
+    "Miscellaneous"
+    [ testProperty
+        "smallHittingSet produces hitting sets"
+        (prop_hittingSet @QC.OrdA)
+    ]
+
+prop_hittingSet :: Ord a => [Set a] -> Property
+prop_hittingSet ss = not (any S.null ss) ==> isHittingSet (smallHittingSet ss) ss
+
+isHittingSet :: (Foldable t, Ord a) => Set a -> t (Set a) -> Bool
+isHittingSet hs = not . any (S.null . S.intersection hs)
+
+-- This property does *not* hold (and should not, because the problem
+-- of producing a minimal hitting set is NP-hard), but we can use it
+-- to generate counterexamples.
+prop_hittingSetMinimal :: [Set El] -> Property
+prop_hittingSetMinimal ss = not (any S.null ss) ==> all ((S.size hs <=) . S.size) allHittingSets
```

Comment:
Nice property test! :+1:

Do you think Quickcheck would find the counterexample with real sets of capabilities that entities have, or would that take too long?

Reply:
It would take way too long, I think.  Also, I'm not sure that any counterexamples actually exist with the real sets of capabilities that entities currently have.